florist/api/launch.py

```diff
@@ -0,0 +1,140 @@
+    server_address: str,
+    n_server_rounds: int,
+) -> None:
+    """
+    Function to start server that executes inside process spawned by launch_server.
+    Redirects logging to console, stdout and stderr to file.
+
+    Args:
+        server_constructor (Callable[FlServer]): Callable that constructs FL server.
+        server_address (str): String of <IP>:<PORT> to make server available.
+        n_server_rounds (str): The number of rounds to perform FL
+    """
+    log_file_name = "server.out"
+    redirect_logging_from_console_to_file(log_file_name)
+    log_file = open(log_file_name, "a")
+    # Send remaining ouput (ie print) from stdout and stderr to file
+    sys.stdout = sys.stderr = log_file
+    server = server_constructor()
+    fl.server.start_server(
+        server=server,
+        server_address=server_address,
+        config=ServerConfig(num_rounds=n_server_rounds),
+    )
+    server.metrics_reporter.dump()
+    log_file.close()
+
+
+def start_client(client: BasicClient, server_address: str) -> None:
+    """
+    Function to start client that executes inside process spawned by launch_client.
+    Redirects logging to console, stdout and stderr to file.
+
+    Args:
+        client (BasicClient): BasicClient instance to launch.
+        server_address (str): String of <IP>:<PORT> the server is available.
+    """
+    log_file_name = f"client_{str(os.getpid())}.out"
+    redirect_logging_from_console_to_file(log_file_name)
+    log_file = open(log_file_name, "a")
+    # Send remaining ouput (ie print) from stdout and stderr to file
+    sys.stdout = sys.stderr = log_file
+    fl.client.start_numpy_client(server_address=server_address, client=client)
+    client.shutdown()
+    log_file.close()
+
+
+def launch_server(
+    server_constructor: Callable[..., FlServer],
+    server_address: str,
+    n_server_rounds: int,
+    seconds_to_sleep: int = 10,
+) -> Process:
+    """
+    Function to that spawns a process that starts FL server.
+
+    Args:
+        server_constructor (Callable[FlServer]): Callable that constructs FL server.
+        server_address (str): String of <IP>:<PORT> to make server available.
+        n_server_rounds (str): The number of rounds to perform FL.
+        seconds_to_sleep (int): The number of seconds to sleep before launching server.
+
+    Returns:
+        Process: The process running the FL server.
+    """
+    server_process = Process(target=start_server, args=(server_constructor, server_address, n_server_rounds))
+    server_process.start()
+    time.sleep(seconds_to_sleep)
+    return server_process
+
+
+def launch_client(client: BasicClient, server_address: str) -> None:
+    """
+    Function to that spawns a process that starts FL client.
+
+    Args:
+        client (BasicClient): BasicClient instance to launch.
+        server_address (str): String of <IP>:<PORT> to make server available.
+    """
+    client_process = Process(target=start_client, args=(client, server_address))
```

Comment:
It's interesting that you can pass whole objects or even functions to this and it works, makes things a lot easier.

Reply:
Yeah from what I can see, passing objects is safe as long as they are not or do not contain a thread.RLock (I ran into that trying to instantiate server than pass to the server process. So thats why I took the approach of passing a constructor instead. 

It is worth noting that depending on how the underlying process is started (spawn vs fork), we have slightly different access to object defined in parent process from child process. Spawn (which is default for mac) will only inherit the resources from the parent process that are required to run the process instance while fork creates a copy of the parent process so it has access to all the resources defined in parent.

I could be wrong but thats what I have gathered so far

Reply:
Great, thanks for looking into that. Spawn sounds god, I think keeping it as isolated as possible would be better in the long term when we need to have an implementation of this for Slurm as well. But this is good enough for now, we will cross that bridge when we get there.